### ISSUE TITLE
Strict OSM node ID numbering inside cucumber test scenarios

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#4120]
Our cucumber test scenarios generate `.osm.xml` files, which are then processed by OSRM, and test are run.

When we generate the OSM XML, the code assigns ID values to every node and way that's generated (for example, [here](https://github.com/Project-OSRM/osrm-backend/blob/master/features/support/data.js#L43) and [here](https://github.com/Project-OSRM/osrm-backend/blob/master/features/support/data.js#L105).

Some of our traffic tests depend on specific ID values being assigned - the CSV data for traffic updates uses the ID values to find connected nodes.

Currently, we have implicit numbering of nodes - the ID assigned depends on the order in which nodes are added to the XML, and *that* depends on the shape of the map drawn in the cucumber test, or the order in which the nodes are described in the cucumber test, or the order in which the JS code extracts the data from a hashmap.

Basically, it's not consistent, and this makes writing tests failure prone.

We should change this code and move to a fixed scheme.  The most obvious choice is:

| Node name | ID |
| ---- | ---- |
| a | 1 |
| b | 2 |
| ... | ... |
| z | 26 |
| A | 27 |
| ... | ... |
| Z | 52 |

This is roughly compatible with existing tests, and gives us 52 nodes to make test maps out of (anyone who makes a test map with more than 52 maps should get their head checked).

**Comment by emiltin** [comment]
there's also the current option of explicitly indicating node ids with an 'id' column
